src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.Worker.cs

```diff
@@ -0,0 +1,499 @@
+                    {
+                        parseOptions = parseOptions.WithDocumentationMode(DocumentationMode.Parse);
+                    }
+
+                    // add all the extra options that are really behavior overrides
+                    var metadataService = GetWorkspaceService<IMetadataService>();
+                    var compilationOptions = commandLineArgs.CompilationOptions
+                        .WithXmlReferenceResolver(new XmlFileResolver(projectDirectory))
+                        .WithSourceReferenceResolver(new SourceFileResolver(ImmutableArray<string>.Empty, projectDirectory))
+                        // TODO: https://github.com/dotnet/roslyn/issues/4967
+                        .WithMetadataReferenceResolver(new WorkspaceMetadataFileReferenceResolver(metadataService, new RelativePathResolver(ImmutableArray<string>.Empty, projectDirectory)))
+                        .WithStrongNameProvider(new DesktopStrongNameProvider(commandLineArgs.KeyFileSearchPaths))
+                        .WithAssemblyIdentityComparer(DesktopAssemblyIdentityComparer.Default);
+
+                    var documents = CreateDocumentInfos(projectFileInfo.Documents, projectId, commandLineArgs.Encoding);
+                    var additionalDocuments = CreateDocumentInfos(projectFileInfo.AdditionalDocuments, projectId, commandLineArgs.Encoding);
+                    CheckForDuplicateDocuments(documents, additionalDocuments, projectPath, projectId);
+
+                    var analyzerReferences = ResolveAnalyzerReferences(commandLineArgs);
+
+                    var resolvedReferences = await ResolveReferencesAsync(projectId, projectFileInfo, commandLineArgs, cancellationToken).ConfigureAwait(false);
+
+                    return ProjectInfo.Create(
+                        projectId,
+                        version,
+                        projectName,
+                        assemblyName,
+                        language,
+                        projectPath,
+                        outputFilePath: projectFileInfo.OutputFilePath,
+                        outputRefFilePath: projectFileInfo.OutputRefFilePath,
+                        compilationOptions: compilationOptions,
+                        parseOptions: parseOptions,
+                        documents: documents,
+                        projectReferences: resolvedReferences.ProjectReferences,
+                        metadataReferences: resolvedReferences.MetadataReferences,
+                        analyzerReferences: analyzerReferences,
+                        additionalDocuments: additionalDocuments,
+                        isSubmission: false,
+                        hostObjectType: null);
+                });
+            }
+
+            private static VersionStamp GetProjectVersion(string projectFilePath)
+            {
+                if (!string.IsNullOrEmpty(projectFilePath) && File.Exists(projectFilePath))
```

Comment:
>if (!string.IsNullOrEmpty(projectFilePath [](start = 16, length = 41)

In what cases would you expect this to be null / empty? The one callsite to GetProjectVersion didn't seem to have an obivous case where this could occur.

Reply:
Looking more closely, you're right. This check is unnecessary. By the time we call this method, the path will have already been validated and errors would have been reported if it was invalid or didn't exist. Cleaning this up...